core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1302,8 +1302,18 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
         )
       }
     }
+  }
 
+  private case class memoVal[T](var value: Option[T], val fn: () => T) {
+    def get(): T = {
+      if (value.isEmpty) {
+        value = Some(fn())
+      }
+      value.get
+    }
   }
+  private var externalRef = memoVal[Boolean](None, () => elements.exists(_._2._id < _id))
+  private[chisel3] def hasExternalRef(): Boolean = externalRef.get()
```

Comment:
```suggestion
  lazy val hasExternalRef: Boolean = this.elements.exists(_._2._id < _id)
```
Scala actually has a built-in for this sort of thing 🙂 This should also use less memory because it only adds a 1 byte field (and uses 1 extra bit in the lazy val mask that already exists because we have other lazy vals in Bundles). Your approach adds a 4-byte field pointing to a 24-40 byte object. This might actually suggest rerunning the memory use check because 28-bytes per Bundle is non-trivial.